### PR TITLE
simdutf: only use `libiconv` on macOS

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Try fixing Linux build.

### CODE REVIEW

File: Formula/simdutf.rb [hunk 1]
```diff
@@ -18,9 +18,12 @@ class Simdutf < Formula
   depends_on "cmake" => :build
   depends_on "python@3.11" => :build
   depends_on "icu4c"
-  depends_on "libiconv"
   depends_on macos: :catalina
 
+  on_macos do
+    depends_on "libiconv"
+  end
```
Comment (carlocab) [inline Formula/simdutf.rb:25]: Seems weird to even be using the formula at all.

Comment (cho-m) [reply Formula/simdutf.rb:25]: I tried locally and system `libiconv` causes some issue when linking Homebrew's `icu4c`.

However, I think `icu4c` is only used when building benchmark program (which also has a requirement on building tests).

Could disable benchmark+testing, but will need a new test as `simdutf` is running benchmark.

Comment (carlocab) [reply Formula/simdutf.rb:25]: Works completely fine for me locally though. Weird.

Comment (cho-m) [reply Formula/simdutf.rb:25]: May actually be a local issue when testing via Rosetta as it seems to find my ARM libraries. Can try dropping `libiconv`. Still need to figure out Linux test failure.

Comment (carlocab) [reply Formula/simdutf.rb:25]: Removing it seems to work fine at #123171.

**Comment by cho-m** [comment]
Closing for now as `libiconv` part is done. Still want to get a Linux bottle but may need to try locally to figure out current failure.

**Pull request closed without merging**